pkgs/os-specific/linux/kernel/sunlight-kernels.nix

```diff
@@ -0,0 +1,70 @@
+{ lib, stdenv, fetchFromGitHub, buildLinux, ... } @ args:
+
+let
+  ltsVariant = {
+    version = "6.1.21";
+    hash = "sha256-IiKQxYN73DVHpjDtQ43+wlRFkNDNALAxiZSwvEUfQjI=";
+    variant = "lts";
+  };
+
+  sunlightKernelFor = { version, suffix ? "sunlight1", hash, variant }: buildLinux (args // rec {
+    inherit version;
+    modDirVersion = lib.versions.pad 3 "${version}-${suffix}";
+
+    src = fetchFromGitHub {
+      owner = "sunlightlinux";
+      repo = "linux-sunlight";
+      rev = modDirVersion;
+      inherit hash;
+    };
+
+    structuredExtraConfig = with lib.kernel; {
+      # Expert option for built-in default values.
+      GKI_HACKS_TO_FIX = yes;
+
+      # AMD P-state driver.
+      X86_AMD_PSTATE = lib.mkOverride 60 yes;
+      X86_AMD_PSTATE_UT = no;
+
+      # Google's BBRv2 TCP congestion Control.
+      TCP_CONG_BBR2 = yes;
+      DEFAULT_BBR2 = yes;
+
+      # FQ-PIE Packet Scheduling.
+      NET_SCH_DEFAULT = yes;
+      DEFAULT_FQ_PIE = yes;
+
+      # Futex WAIT_MULTIPLE implementation for Wine / Proton Fsync.
+      FUTEX = yes;
+      FUTEX_PI = yes;
+
+      # WineSync driver for fast kernel-backed Wine.
+      WINESYNC = module;
+
+      # Preemptive Full Tickless Kernel at 833Hz.
+      LATENCYTOP = yes;
+
+      PREEMPT = lib.mkOverride 60 yes;
+      PREEMPT_VOLUNTARY = lib.mkForce no;
+      PREEMPT_NONE = lib.mkForce no;
+
+      # 833 Hz is alternative to 1000 Hz.
+      # Selected value for a balance between latency, performance and low power consumption.
+      HZ = freeform "833";
+      HZ_833 = yes;
+      HZ_1000 = no;
+
+      SCHEDSTATS = lib.mkOverride 60 yes;
+    };
+
+    extraMeta = {
```

Comment:
can you add `homepage`?

Reply:
I don't have a homepage or website at the moment. I am also working on this aspect.

Reply:
if you have a readme in the repo that explains what it is etc, that is also fine

Reply:
I will try to add more information about it along the way.